siem_integrations/README.md

```diff
@@ -28,8 +28,92 @@ In order to utilize splunk2kafka, a [running Kafka instance](https://kafka.apach
+4. CLX Query Service Runner usage.
+
+    ```aidl
+    bash clx_query_service/bin/start_service.sh --help
+    ```
+    ``` 
+    Usage: clx_query_service/bin/start_service.sh [POS]... [ARG]...
+    Example-1: bash clx_query_service/bin/start_service.sh -p 8998 -w 2 -t 60
+    Example-2: bash clx_query_service/bin/start_service.sh --port 8990 --workers 4 --timeout 10
+    
+    CLX Query Service Runner
+    
+    Positional:
+      -p, --port          Port number
+      -w, --workers       Number of workers
+      -t, --timeout       Application time out
+    
+      -h, --help          Print this help
+    ```
+5. Run Gunicorn wrapped CLX Query Service application as daemon process using Supervisor.
+   1. Update `command` property in the `clx_query_service/conf/clx_query_service.conf` as per the user requirements. Such as service binding `port`, number of `workers` and request `timeout` seconds.
+   2. Copy update configuration file to supervisor location.
+        ```aidl
+        cp clx_query_service/conf/clx_query_service.conf /etc/supervisor
+        ```
+   3. Add configuration file to Supervisord
+        ```aidl
+        supervisord -c /etc/supervisor/clx_query_service.conf
+        ```
+   4.  Add configuration file to Supervisorctl.
+        ```aidl
+        supervisorctl -c /etc/supervisor/clx_query_service.conf
+        ```
+   5.  Start CLX Query Service from Supervisorctl CLI.
+        ```aidl
+        start clx_query_service
+        ```
+### Install and Run CLX Query
+
+1. To establish communication between CLX Query and CLX Query Service, update the configuration file `clx query/default/clx_query_setup.conf` with the CLX Query Service hostname and the port number. The value of the `hostname` property should be the same as the IP used in the porperty `ALLOWED HOSTS` of `clx_query_service/clx_query_service/settings.py`. The value of the `port` property should be the same as the number used in the `command` property of `clx_query_service/conf/clx_query_service.conf`.
+
+2. Copy CLX Query to splunk apps directory.
+    ```aidl
+    cp -R clx_query splunk/etc/apps
+    ```
+3. Restart splunk application server to take effect on changes.
+    ```aidl
+    ./splunk/bin/splunk restart
+    ``` 
+4. Login to Splunk GUI and launch CLX Query application `Apps> Manage Apps> Clx Query> Launch App` 
+5. Run sample query.
+    ```
+    | clx query="SELECT genres, title, count(user_id) as user_cnt, avg(rating) as avg_rating from (SELECT main.movies.title as title, main.movies.genres as genres, main.ratings.userId as user_id, main.ratings.rating as rating FROM main.movies INNER JOIN main.ratings ON (main.ratings.movieId = main.movies.movieId) WHERE main.ratings.rating > 2.5) as tmp GROUP BY genres, title ORDER BY user_cnt DESC, avg_rating DESC"
+    ```   
+   
```

Comment:
It might be good to include a brief 1-2 sentences here about what the user should expect to see after they run the query. You could also put a screenshot of what it's supposed to look like as well. I'll leave that up to you. If you do use a screenshot, remember no IP addresses, hostnames, etc.